libs/components/popovers/src/lib/modules/dropdown/dropdown.component.ts

```diff
@@ -183,45 +232,50 @@ export class SkyDropdownComponent implements OnInit, OnDestroy {
   public menuContainerTemplateRef: TemplateRef<unknown> | undefined;
 
   @ViewChild('triggerButton', {
-    read: ElementRef,
-    static: true,
+    read: SkyDropdownTriggerBaseDirective,
   })
-  public set triggerButton(value: ElementRef | undefined) {
+  public set triggerButton(value: SkyDropdownTriggerBaseDirective | undefined) {
     this.#_triggerButton = value;
     this.#addEventListeners();
+    this.#updateTrigger();
   }
 
-  public get triggerButton(): ElementRef | undefined {
-    return this.#_triggerButton;
+  public get triggerButton():
+    | SkyDropdownTriggerBaseDirective
+    | SkyDropdownTriggerDirective
+    | undefined {
+    return this.#_customTriggerButton ?? this.#_triggerButton;
   }
 
-  protected destroyRef = inject(DestroyRef);
+  @ContentChild(SkyDropdownTriggerDirective)
+  public set customTriggerButton(
+    value: SkyDropdownTriggerDirective | undefined,
+  ) {
+    this.#_customTriggerButton = value;
+    this.#addEventListeners();
+    this.#updateTrigger();
+  }
+
+  protected screenReaderLabelContextMenuId = this.#idSvc.generateId();
 
   #affixer: SkyAffixer | undefined;
-  #overlay: SkyOverlayInstance | undefined;
   #ngUnsubscribe = new Subject<void>();
+  #triggerUnsubscribe = new Subject<void>();
+  #overlay: SkyOverlayInstance | undefined;
   #positionTimeout: number | undefined;
 
-  readonly #affixService = inject(SkyAffixService);
-  readonly #changeDetector = inject(ChangeDetectorRef);
-  readonly #environmentInjector = inject(EnvironmentInjector);
-  readonly #overlayService = inject(SkyOverlayService);
-  readonly #themeSvc = inject(SkyThemeService, { optional: true });
-  readonly #zIndex = inject(SKY_STACKING_CONTEXT, { optional: true })?.zIndex;
-  readonly #contentInfoProvider = inject(SkyContentInfoProvider, {
-    optional: true,
-  });
-
   #_buttonStyle = DEFAULT_BUTTON_STYLE;
   #_buttonType = DEFAULT_BUTTON_TYPE;
+  #_customTriggerButton: SkyDropdownTriggerDirective | undefined;
+  #_disabled: boolean | undefined;
```

Comment:
Not a huge deal, but this value was set to a hard `false`, before. Also, since we're now using a setter on the input, we'll need to add `@default false` to the JSDoc comment since TypeDoc can't infer the default value anymore.

Reply:
I'm not sure why, but when I moved this over to a directive and the value was `false`, it rendered `disabled="false"` in the HTML which resulted in the field always being disabled. I removed the default `false` value and added additional logic to flip to `undefined` when false so the attribute wouldn't be added at all when false. Since I have that second check I can put the default value back. However, the `@default false` is still in the JSDoc comment on the setter.